payjoin/src/receive/v1/mod.rs

```diff
@@ -1009,6 +1044,34 @@ pub(crate) mod test {
         };
     }
 
+    #[test]
+    fn test_avoid_uih_one_input() {
+        let proposal = unchecked_proposal_from_test_vector().unwrap();
+        let proposal_psbt = Psbt::from_str(RECEIVER_INPUT_CONTRIBUTION).unwrap();
+        let input = InputPair {
+            txin: proposal_psbt.unsigned_tx.input[1].clone(),
+            psbtin: proposal_psbt.inputs[1].clone(),
+        };
+        let input_iter = [input.clone()].into_iter();
+        let mut payjoin = wants_outputs_from_test_vector(proposal)
+            .unwrap()
+            .commit_outputs()
+            .contribute_inputs(vec![input])
```

Comment:
is this clone necessary? This should work:

```suggestion
        let input_iter = [input].into_iter();
        let mut payjoin = wants_outputs_from_test_vector(proposal)
            .unwrap()
            .commit_outputs()
            .contribute_inputs(input_iter)
```